contracts/src/Utilities/ValidatorSetSig.sol

```diff
@@ -83,31 +103,35 @@ contract ValidatorSetSig is ReentrancyGuard {
             "ValidatorSetSig: non-zero originSenderAddress"
         );
 
-        ValidatorSetSigMessage memory validatorSetSigMessage;
-        (validatorSetSigMessage) = abi.decode(message.payload, (ValidatorSetSigMessage));
-        require(
-            validatorSetSigMessage.validatorSetSigAddress == address(this),
-            "ValidatorSetSig: invalid validatorSetSigAddress"
-        );
-        require(
-            validatorSetSigMessage.targetBlockChainID == blockchainID,
-            "ValidatorSetSig: invalid targetBlockChainID"
-        );
+        ValidatorSetSigMessage memory validatorSetSigMessage =
+            abi.decode(message.payload, (ValidatorSetSigMessage));
 
-        require(
-            nonces[validatorSetSigMessage.targetContractAddress] + 1 == validatorSetSigMessage.nonce,
-            "ValidatorSetSig: invalid nonce"
-        );
+        validateMessage(validatorSetSigMessage);
 
         nonces[validatorSetSigMessage.targetContractAddress] = validatorSetSigMessage.nonce;
 
         (bool success,) =
         // solhint-disable-next-line avoid-low-level-calls
-         validatorSetSigMessage.targetContractAddress.call(validatorSetSigMessage.txPayload);
+         validatorSetSigMessage.targetContractAddress.call(validatorSetSigMessage.payload);
```

Comment:
Can you add a comment explaining that since the caller is expected to have full control over the contract called (since it requires a threshold of validators to agree), we don't need to protect against return bomb vectors. For context, we _do_ have to protect against that in `TeleporterMessenger` as well as `TeleporterTokenBridge`: https://github.com/ava-labs/teleporter-token-bridge/blob/afdb3c21691c0ddb389e2a2ae359a43074249ce1/contracts/src/utils/CallUtils.sol#L24

Reply:
is there downside to using the same assembly execute call?